lib/auth/httpfallback.go

```diff
@@ -831,3 +831,51 @@ func (c *Client) GenerateHostCerts(ctx context.Context, req *proto.HostCertsRequ
 
 	return &certs, nil
 }
+
+// DELETE IN 9.0.0, to remove fallback and grpc call is already defined in api/client/client.go
+//
+// CreateAuthenticateChallenge creates and returns MFA challenges for a users registered MFA devices.
+func (c *Client) CreateAuthenticateChallenge(ctx context.Context, req *proto.CreateAuthenticateChallengeRequest) (*proto.MFAAuthenticateChallenge, error) {
+	switch resp, err := c.APIClient.CreateAuthenticateChallenge(ctx, req); {
+	case err == nil:
+		return resp, nil
+	case !trace.IsNotImplemented(err):
+		return nil, trace.Wrap(err)
+	}
+
+	// HTTP fallback for auth version <7.x
+	out, err := c.PostJSON(
+		c.Endpoint("u2f", "users", req.GetUserCredentials().GetUsername(), "sign"),
+		signInReq{
+			Password: string(req.GetUserCredentials().GetPassword()),
+		},
+	)
+
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	var jsonChal *MFAAuthenticateChallenge
+	if err := json.Unmarshal(out.Bytes(), &jsonChal); err != nil {
+		return nil, err
+	}
+
+	// Convert JSON back to proto.
+	// Webauthn is not handled here b/c the feature
+	// does not exist for auth version <7.x
+	protoChal := &proto.MFAAuthenticateChallenge{}
+
+	if jsonChal.TOTPChallenge {
+		protoChal.TOTP = &proto.TOTPChallenge{}
+	}
+
+	for _, u2fChal := range jsonChal.U2FChallenges {
+		protoChal.U2F = append(protoChal.U2F, &proto.U2FChallenge{
+			KeyHandle: u2fChal.KeyHandle,
+			Challenge: u2fChal.Challenge,
+			AppID:     u2fChal.AppID,
+		})
+	}
+
+	return protoChal, nil
```

Comment:
Use MakeAuthenticateChallenge here?

Reply:
ah, the `MakeAuthenticateChallenge` is used to convert proto to JSON, this block is doing the opposite